moving ParseSelector to Parse for labels only.

sdminonne opened the pull request:
@smarterclayton @thockin 

In this PR `labels.ParseSelector` is systematically replaced with `labels.Parse`. For the moment I didn't create the `fields` package as agreed in https://github.com/GoogleCloudPlatform/kubernetes/issues/341#issuecomment-75712472

In this PR I've also slightly modified `selector.go` following @smarterclayton suggestion and [this](https://code.google.com/p/go-wiki/wiki/CodeReviewComments).

sdminonne committed:
moving ParseSelector to Parse for labels only.

sdminonne commented:
file plugin/pkg/scheduler/factory/factory.go try to parse `DesiredState.Host!=`. Restored the old `ParseSelector` but I'm not sure about. Is this a `field`?

bgrant0607 commented:
Yes, DesiredState.Host is a v1beta1 pod field.

sdminonne commented:
Thanks

brendandburns commented:
We use labels all over the place in e2e tests in hack/e2e-suite/*.sh

Can you update the calls there too?

Also, needs rebase.

thockin commented:
I am not going to have time to review this in the next week.  Brian - who
else can we loop in on this area?

On Thu, Feb 26, 2015 at 2:58 AM, Salvatore Dario Minonne <
notifications@github.com> wrote:

> @smarterclayton <https://github.com/smarterclayton> @thockin
> <https://github.com/thockin>
>
> In this PR labels.ParseSelector is systematically replaced with
> labels.Parse. For the moment I didn't create the fields package as agreed
> in #341 (comment)
> <https://github.com/GoogleCloudPlatform/kubernetes/issues/341#issuecomment-75712472>
>
> In this PR I've also slightly modified selector.go following
> @smarterclayton <https://github.com/smarterclayton> suggestion and this
> <https://code.google.com/p/go-wiki/wiki/CodeReviewComments>.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/GoogleCloudPlatform/kubernetes/pull/4835
> Commit Summary
>
>    - moving ParseSelector to Parse for labels only.
>
> File Changes
>
>    - *M* hack/test-cmd.sh
>    <https://github.com/GoogleCloudPlatform/kubernetes/pull/4835/files#diff-0>
>    (12)
>    - *M* pkg/apiserver/resthandler.go
>    <https://github.com/GoogleCloudPlatform/kubernetes/pull/4835/files#diff-1>
>    (2)
>    - *M* pkg/apiserver/watch.go
>    <https://github.com/GoogleCloudPlatform/kubernetes/pull/4835/files#diff-2>
>    (2)
>    - *M* pkg/client/cache/listwatch_test.go
>    <https://github.com/GoogleCloudPlatform/kubernetes/pull/4835/files#diff-3>
>    (2)
>    - *M* pkg/client/client_test.go
>    <https://github.com/GoogleCloudPlatform/kubernetes/pull/4835/files#diff-4>
>    (16)
>    - *M* pkg/client/request.go
>    <https://github.com/GoogleCloudPlatform/kubernetes/pull/4835/files#diff-5>
>    (2)
>    - *M* pkg/client/request_test.go
>    <https://github.com/GoogleCloudPlatform/kubernetes/pull/4835/files#diff-6>
>    (2)
>    - *M* pkg/kubectl/resource/builder.go
>    <https://github.com/GoogleCloudPlatform/kubernetes/pull/4835/files#diff-7>
>    (2)
>    - *M* pkg/labels/selector.go
>    <https://github.com/GoogleCloudPlatform/kubernetes/pull/4835/files#diff-8>
>    (228)
>    - *M* pkg/labels/selector_test.go
>    <https://github.com/GoogleCloudPlatform/kubernetes/pull/4835/files#diff-9>
>    (59)
>    - *M* pkg/registry/generic/registry_test.go
>    <https://github.com/GoogleCloudPlatform/kubernetes/pull/4835/files#diff-10>
>    (2)
>    - *M* pkg/registry/pod/etcd/etcd_test.go
>    <https://github.com/GoogleCloudPlatform/kubernetes/pull/4835/files#diff-11>
>    (2)
>    - *M* plugin/pkg/scheduler/factory/factory.go
>    <https://github.com/GoogleCloudPlatform/kubernetes/pull/4835/files#diff-12>
>    (2)
>
> Patch Links:
>
>    - https://github.com/GoogleCloudPlatform/kubernetes/pull/4835.patch
>    - https://github.com/GoogleCloudPlatform/kubernetes/pull/4835.diff
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/4835>.
>

bgrant0607 commented:
@nikhiljindal is working on versioning of field selectors, which is related.

sdminonne commented:
@nikhiljindal @brendanburns @thockin I've rebased.

I've also modified e2e tests but I couldn't run them  to verify them. (I've no gcloud account) and vagrant env is broken. For the moment I'm updating the new code. Working on my side to run the e2e.

nikhiljindal commented on pkg/client/request.go:
```diff
@@ -236,7 +236,7 @@ func (r *Request) ParseSelectorParam(paramName, item string) *Request {
 		return r
 	}
-	sel, err := labels.ParseSelector(item)
+	sel, err := labels.Parse(item)
```
We need to use ParseSelector when paramName = "fields" and Parse when paramName = "labels".

sdminonne replied:
Thanks

nikhiljindal commented on hack/e2e-suite/liveness.sh:
```diff
@@ -51,7 +51,7 @@ function teardown() {
 
 function waitForNotPending() {
-  pod_id_list=$(${KUBECTL} get pods -o template '--template={{range.items}}{{.id}} {{end}}' -l test=liveness)
+  pod_id_list=$(${KUBECTL} get pods -o template '--template={{range.items}}{{.id}} {{end}}' -l'test in (liveness)')
```
Why is this change required?
IIUC, both "test=liveness" and "test in (liveness)" will work, right?

sdminonne replied:
yeah. I'll rollback this

nikhiljindal commented:
Can you please rebase.
We should be able to delete labels.ParseSelector method after rebase. 
And then you can move labels.ParseAndTransformSelector to fields.ParseSelector whenever you plan to do that as per https://github.com/GoogleCloudPlatform/kubernetes/issues/341#issuecomment-75712472

sdminonne commented:
@nikhiljindal working on this

sdminonne commented:
@nikhiljindal rebased. How do you want to proceed to replace ParseSelector with ParseAndTransformSelector?

bgrant0607 commented:
@sdminonne I'd like to get this merged. Did you pick up some other files by mistake, such as hack/e2e-suite/pd.sh?

bgrant0607 commented:
pd.sh was removed:
https://github.com/GoogleCloudPlatform/kubernetes/commit/7bb7ba65774165bcd72acf308420274cf405431f

Please also check hack/e2e-suite/monitoring.sh and hack/lib/test.sh. Those changes don't look relevant to this PR.

Maybe try to rebase again?

When you get this cleaned up, I'd like to merge and iterate from there.

sdminonne commented:
My bad. I'll fix it asap.
Not  at my desk now.

Dario

@sdminonne <https://github.com/sdminonne> I'd like to get this merged. Did
you pick up some other files by mistake, such as hack/e2e-suite/pd.sh?

—
Reply to this email directly or view it on GitHub
<https://github.com/GoogleCloudPlatform/kubernetes/pull/4835#issuecomment-77422615>
.

bgrant0607 commented:
At some point -- doesn't have to be this PR, please update labels.md with the extended syntax.

sdminonne commented:
@bgrant0607 PTAL
1) removed hack/e2e-suite/pd.sh
2) hack/e2e-suite/monitoring.sh get rid of spurious modif
3) hack/lib/test.sh needs to be there since hack/test-cmd.sh has been modified with the new syntax and we now have `'-l name in (valid-pod)'` in single quote.

I'll modify labels.md tomorrow

bgrant0607 commented:
LGTM
